### PR TITLE
[docs] Describe restrictions in configuring ECS serviceDef & taskDef

### LINKED ISSUE #4961: [docs] Describe which params are actually valid for servicedef/taskdef of ECS
**What would you like to be added**:

Add docs on which parameters of taskDefinition and serviceDefinition can be actually configured.

**Why is this needed**:

Actually, we can't configure some parameters because they're ignored when creating a service or taskDefinition.
(e.g. `loadbalancers` in servicedef)
For users, that's confusing and noisy for troubleshooting.

- CreateService
https://github.com/pipe-cd/pipecd/blob/b20163b70902d4f0f6698ff85f8ac427d61359f8/pkg/app/piped/platformprovider/ecs/client.go#L90-L106

- RegisterTaskDefinition
https://github.com/pipe-cd/pipecd/blob/b20163b70902d4f0f6698ff85f8ac427d61359f8/pkg/app/piped/platformprovider/ecs/client.go#L150-L163

cf. The current doc does not mention the limitation.
https://pipecd.dev/docs-v0.47.x/user-guide/configuration-reference/#ecs-application

**TBD**
I wonder where to describe that.

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

as title.

**Which issue(s) this PR fixes**:

Fixes #4961

**Does this PR introduce a user-facing change?**: N/A

- **How are users affected by this change**: N/A
- **Is this breaking change**: N/A
- **How to migrate (if breaking change)**: N/A

**How I listed**

- Read AWS docs
  - servicedef: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service_definition_parameters.html
  - taskdef: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html
- Read PipeCD code: ECS>client.go
  - https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/platformprovider/ecs/client.go
- Checked AWS SDK's structs

### CODE REVIEW

**Review by khanhtc1202 — APPROVED** [review]
Nice improvement 👍

**Review by ffjlabo — APPROVED** [review]
Thanks!

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -505,12 +505,32 @@ One of `yamlField` or `regex` is required.
...
+- `taskDefinition` is not supported. PipeCD uses the definition in `taskDefinitionFile` in [ECSDeploymentInput](#ecsdeploymentinput).
+
+### Restrictions of Task Definition
+
+There are some restrictions in configuring a task definition file.
+
+- `placementConstraints` is not supported.
+- `proxyConfiguration` is not supported.
+- `tags` is not supported.
+
```
Comment (Warashi) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:533]: There are notes describing some restrictions on [Configuring ECS application page](https://pipecd.dev/docs-v0.47.x/user-guide/managing-application/defining-app-configuration/ecs/#note).
Please move these notes here and mention them on that page.

**Review by Warashi — COMMENTED** [review]
Some documents have similar notes with this PR, so let's merge.

**Pull request merged**